src/main/kotlin/com/learnspigot/bot/entity/UserProfile.kt

```diff
@@ -27,6 +27,21 @@ data class UserProfile(
         reputation.add(point)
         acknowledgeRep(point, leaderboardManager, guild)
     }
+
+    fun addRep(leaderboardManager: LeaderboardManager, guild: Guild, reason: String, count: Int) {
+        for (rep in count.downTo(1)) {
+            reputation.add(ReputationPoint(System.currentTimeMillis(), null, null))
+            leaderboardManager.updateLeaderboardMessages()
+        }
+        val channel = guild.getTextChannelById(System.getenv("SUPPORT_CHANNEL_ID"))!!
+        if (reputationPing)
+            channel.sendMessage("<@${id}>").queue{ it.delete().queue{} }
+        channel.sendMessageEmbeds(Embed {
+            title = "Reputation Added!"
+            description = "<@$id> has received $count rep point${if (count > 1) "s" else ""} from making a $reason"
+            color = EMBED_COLOR
+        }).queue()
+    }
```

Comment:
refactor the acknowledge function to something to do with help posts, and then move this into another acknowledge related function

Reply:
it is already a refactored function similar as acknowledge rep? it is just with a loop which it doesnt have